### PR TITLE
Use setuptools entry_points for command line scripts

### PR DESCRIPTION
**Pull request by cdeil**
Use setuptools entry_points for command line scripts ... following the example set by Astropy core:
https://github.com/astropy/astropy/pull/3314, https://github.com/astropy/astropy/pull/3271

@embray @saimn Looks OK?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -78,6 +78,12 @@
 package_info['package_data'].setdefault(PACKAGENAME, [])
 package_info['package_data'][PACKAGENAME].append('data/*')
 
+# Define entry points for command-line scripts
+entry_points = {}
+entry_points['console_scripts'] = [
+    'astropy-package-template-example = packagename.example_mod:main',
+]
```
Comment (embray) [inline setup.py]: Is there a reason the empty dict is created first?

Comment (cdeil) [inline setup.py]: Not really ... If you find it simpler when combined into one statement I'll make the change.

Comment (embray) [inline setup.py]: Eh.  I think it's fine for now :)

**Comment by astrofrog** [comment]
@cdeil - please include an entry in ``TEMPLATE_CHANGES.md``

**Comment by cdeil** [comment]
@astrofrog - Changelog entry added.

**Comment by saimn** [comment]
Thanks for taking care of this @cdeil :)

**Comment by astrofrog** [comment]
@cdeil - thanks!

**Comment by SaraOgaz** [comment]
Is it just me, or has this made the documentation out of date... I'm trying to follow the astropy page  for setting up an affiliated package (http://docs.astropy.org/en/latest/development/affiliated-packages.html), and it still lists

``scripts/script_example`` as a thing...

**Comment by pllim** [comment]
@SaraOgaz , does look outdated. Feel free to open a PR at Astropy to update it. :smile:

**Comment by SaraOgaz** [comment]
I'll try and do that this week.

**Pull request merged**